src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/loptimizer/heuristics/utils/Var_Analyzer.java

```diff
@@ -0,0 +1,35 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.loptimizer.heuristics.utils;
+
+import org.apache.jena.graph.Node;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class Var_Analyzer {
+
+//    varsTotal: a set of all variables that have been calculated
+//    Add all bound variables to bindings
+    public static int calculateVars( final List<Node> vars, final Set<Node> varsTotal, final Set<Node> bindings ) {
+        varsTotal.addAll( getUniqueNodes(vars) );
```

Comment:
Given that `varsTotal` is a `Set`, there is no need to call `getUniqueNodes` and create yet another `HashSet` there.